lib/ebml/tools.js

```diff
@@ -50,6 +63,161 @@ var tools = {
         }
         buffer[0] = buffer[0] | (1 << (8 - length));
         return buffer;
+    },
+
+    /**
+     * *
+     * concatenate two arrays of bytes
+     * @param {Uint8Array} a1  First array
+     * @param {Uint8Array} a2  Second array
+     * @returns  {Uint8Array} concatenated arrays
+     */
+    concatenate: function(a1, a2) {
+        if (!a1 || a1.byteLength === 0) {
+            return a2;
+        }
+        if (!a2 || a2.byteLength === 0) {
+            return a1;
+        }
+        var result = new Uint8Array(a1.byteLength + a2.byteLength);
+        result.set(a1, 0);
+        result.set(a2, a1.byteLength);
+        a1 = null;
+        a2 = null;
+        return result;
+    },
+
+    /**
+     * get a hex text string from Buff[start,end)
+     * @param {Uint8Array} buff from which to read the string
+     * @param {Number} start, default 0
+     * @param {Number} end, default the whole buffer
+     * @returns {string} the hex string
+     */
+    readHexString: function(buff, start, end) {
+        var result = '';
+
+        if (!start) {
+            start = 0;
+        }
+        if (!end) {
+            end = buff.byteLength;
+        }
+
+        for (var p = start; p < end; p++) {
+            var q = Number(buff[p] & 0xff);
+            result += ("00" + q.toString(16)).substr(-2);
+        }
+        return result;
+    },
+    readUtf8: function(buff) {
+        if (typeof window === 'undefined') {
+            return new Buffer.from(buff).toString("utf8");
```

Comment:
As far as I know (I might be wrong here) the `new` is not needed (or even wrong?) here since `Buffer.from` creates a buffer itself.